app/assets/javascripts/backbone/views/log_modal_view.coffee

```diff
@@ -0,0 +1,45 @@
+class MS.Views.LogModalView extends Backbone.View
+
+  el: '#log-modal'
+
+  events:
+    'click [data-action="submit"]': 'submitForm'
+    'ajax:success': 'ajaxSuccess'
+    'click [data-control]': 'expandContent'
+
+  showEdit: (logId, stepId) ->
+    MS.loadingIndicator.show()
+    @stepId = stepId
+    $.get "/admin/project_logs/#{logId}/edit", (html) =>
+      @replaceContent(html)
+
+  showNew: (stepId) ->
+    MS.loadingIndicator.show()
+    @stepId = stepId
+    $.get '/admin/project_logs/new', step_id: @stepId, (html) =>
+      @replaceContent(html)
+
+  replaceContent: (html) ->
+    @$el.find('.modal-content').html(html)
+    new MS.Views.TranslationsView({
+      el: @$('[data-content-translatable="log"]')
+    })
+    @$el.modal('show')
+    MS.loadingIndicator.hide()
+
+  submitForm: (e) ->
+    e.preventDefault()
+    @$el.find('form').submit()
+    @$el.modal('hide')
+
+  ajaxSuccess: (e, data) ->
+    step = $(".timeline [data-step-id='#{@stepId}']")
+    $(step).find('.step-logs').html(data).addClass('expanded')
+
+  expandContent: (e) ->
+    e.preventDefault()
```

Comment:
Our current expandable function in application view model will not work here, as it works for larger containers containing many items and the expand button contained within that large container. This is similar, but this function controls one specific element at a time and can be used next to another control that also expands a different element.